Adds multiple pages to the Onboarding chapter

ConorOkus opened the pull request:
Adds the following pages:
 - Backing up a recovery phrase 
 - Restoring a wallet 
 - Protecting a wallet 
 - Restoring a wallet

Checkout the preview [here](https://deploy-preview-302--sad-borg-390916.netlify.app/guide/onboarding/introduction/)

Closes the following issues:
https://github.com/BitcoinDesign/Guide/issues/243
https://github.com/BitcoinDesign/Guide/issues/242
https://github.com/BitcoinDesign/Guide/issues/244

Bosch-0 commented on guide/onboarding/automatic-cloud-backup.md:
```diff
@@ -0,0 +1,98 @@
+
+{% include tip/open.html color="green" icon="check" label="Do: Inform your Users" %}
+
+Before automatically backing up a recovery phrase via iCloud, make sure the user knows what they are about to do. Once the backup is complete, provide a summary and some reassurances before moving on. This is a good place for loading interactions and animation.
```
```suggestion
Before automatically backing up a recovery phrase via cloud backup, make sure the user knows what they are about to do. Once the backup is complete, provide a summary and some reassurances before moving on. This is a good place for loading interactions and animation.
```

This sounds better being more generic than specifying a cloud provider.

Bosch-0 commented on guide/onboarding/automatic-cloud-backup.md:
```diff
@@ -0,0 +1,98 @@
+
+{% include tip/open.html color="red" icon="forbid" label="Don’t: Use for High Value Amounts" %}
+
+Automatic cloud backups should not be considered for products with users that are storing high-value amounts. This is because there are potentially more security threat vectors. However, it’s important to note that new users will probably not be storing high values in their wallet, making it an appropriate mechanism to implement in a bitcoin product for beginners.
```
I don't agree with the assumption that new users will not be storing high value amounts. I think this does occur, new users can still own a considerable amount of bitcoin, especially older generations who have the capital to invest but may not be tech savvy enough to look into proper storage. 

To remedy this I think you should mention that it should be communicated to users using cloud backups that it is not recommended to use this backup method for high value amounts.

Bosch-0 commented on guide/onboarding/funding-a-wallet.md:
```diff
@@ -0,0 +1,91 @@
+
+### Earning bitcoin
+
+Some users may choose to accept bitcoin as a form of payment. Many people already started accepting bitcoin for their work. Some employers might also choose to pay part of their staff's salary in bitcoin. This is one way to fill a wallet.
```
```suggestion
Earning bitcoin through exchanging goods and services for it is one way to obtain bitcoin. Some individuals and/or businesses have already started accepting and even paying employers in bitcoin. This is one way to fill a wallet.
```

Bosch-0 commented on guide/onboarding/funding-a-wallet.md:
```diff
@@ -0,0 +1,91 @@
+Cons:
+ - Managing taxes
+ - Difficult to find businesses/individuals willing to pay in bitcoin
+ - Hard to set up recurring paymets
```
```suggestion
 - Hard to set up recurring payments
```

spelling. Also, most exchanges offer auto DCA these days so is this really a con?

ConorOkus replied:
Hmm maybe not, in the UK at least we only have one Bitcoin-only exchange offering reoccurring payments that I know of.

ConorOkus replied:
Maybe I change to "limited options"

Bosch-0 commented on guide/onboarding/funding-a-wallet.md:
```diff
@@ -0,0 +1,91 @@
+
+Cons:
+ - Managing taxes
+ - Difficult to find businesses/individuals willing to pay in bitcoin
```
```suggestion
```

Bosch-0 commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,77 @@
+
+Within your product upon PIN creation, tell your users what a good, secure PIN is (i.e. not their birthday). Recommend against (or even make it impossible) to create a PIN such as 1234, 1111, etc.
+
+{% include tip/close.html %}
```
Something you could add here that I think is useful is having the option to create a 'hidden' wallet for plausible deniability. This is something probably best setup internally as an extra security measure but probably worth mentioning. BlueWallet is an example of a mobile wallet with this feature

ConorOkus replied:
Great idea, thinking about implementing the work Tiana did [here](https://github.com/BitcoinDesign/Guide/pull/112)

ConorOkus replied:
Added as part of the "protecting a wallet" page.

Bosch-0 commented on guide/onboarding/backing-up-a-recovery-phrase.md:
```diff
@@ -0,0 +1,44 @@
+
+# Backing up a recovery phrase
+
+When a user generates a new wallet, it results in creating something called a private key (also known as a recovery phrase). A [recovery phrase]({{ 'guide/glossary/#recovery-phrase' | relative_url }}), is a group of 12 or 24 words that let users recover funds if wallet access is lost. Likewise, users who lose both their recovery phrase and wallet access permanently lose their bitcoin. Arguably, t ...
```
```suggestion
When a user generates a new wallet, it results in creating something called a private key which is how users access their bitcoin. These private keys are often backed up via a recovery phrase. A [recovery phrase]({{ 'guide/glossary/#recovery-phrase' | relative_url }}), is a group of 12 or 24 words that let users recover funds if wallet access is lost. Users who lose both their recovery phrase and wallet access permanently lose their bitcoin. Arguably, this creates one of the toughest design problems within the space.
```

Bosch-0 reviewed (changes requested):
Left some comments, LGTM but overall!

GBKS commented on guide/onboarding/automatic-cloud-backup.md:
```diff
@@ -0,0 +1,98 @@
+   retina = "/assets/images/guide/onboarding/backing-up-a-recovery-phrase/automatic-cloud-backup/cloud-backup-intro@2x.png"
+   alt-text = "Screen explaining automatic cloud backups"
+   width = 250
+   height = 541
```
You can add `class = "shadow"` to these images if you want the drop shadow effect for a bit more contrast.

GBKS replied:
Captions would also be great for these to explain the purpose of each screen.

GBKS replied:
Sorry, should be `layout = "shadow"`.

GBKS commented on guide/onboarding/automatic-cloud-backup.md:
```diff
@@ -0,0 +1,98 @@
+
+# Automatic cloud backup
+
+### How it Works
```
Switch headers on this page to sentence-case for consistency with the rest of the guide.

GBKS commented on guide/onboarding/automatic-cloud-backup.md:
```diff
@@ -0,0 +1,98 @@
+
+To grant users some ownership of their keys, you can create an optional manual backup within settings. A great way to do this is by framing it as an extra layer of security, or a good alternative in the case that the user loses access to their cloud account/if they feel uncomfortable trusting their provider.
+
+In the next section we will show different methods for [manual backups]({{ 'guide/onboarding/backing-up-a-recovery-phrase/manual-backups' | relative_url }}).
```
```suggestion
In the next section we will show different methods for [manual backups]({{ 'guide/onboarding/backing-up-a-recovery-phrase/manual-backup' | relative_url }}).
```
Broken link due to extra "s".

GBKS commented on guide/onboarding/manual-backup.md:
```diff
@@ -0,0 +1,287 @@
+
+### How it Works
+
+As previously stated, onboarding flows that require users to manually back up their seed phrase typically shows users their respective 12-24 word phrase, expecting them to then write it down on a piece of paper and store it in a safe (but memorable location). In the case that a user’s device breaks or  is stolen, the user can recover their funds and wallet by correctly entering their seed phrase. ...
```
`As previously stated` made more sense when this was all one big page, but now it's not so clear. I'd either remove it or turn it into a link to where it was stated before.

GBKS commented on guide/onboarding/manual-backup.md:
```diff
@@ -0,0 +1,287 @@
+
+Lastly, another way to have a user confirm that they recorded their seed phrase correctly is by asking them to select (or manually type out) a random word from their seed phrase. For example, you would ask them to “type out word number 5”, or “type out word number 11”. To ensure maximum security, make sure you do this with 3-4 different words within their phrase. This design is less cumbersome fo ...
+
+
```
Add link to the next page.

GBKS commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,77 @@
+{% include tip/open.html color="green" icon="check" label="Do: Educate on Good Security Practices" %}
+
+Within your product upon PIN creation, tell your users what a good, secure PIN is (i.e. not their birthday). Recommend against (or even make it impossible) to create a PIN such as 1234, 1111, etc.
+
```
Add link to the next page.

pavlenex reviewed (commented):
Great to see the Onboarding section coming together. I've reviewed everything but the `Protecting a wallet` section since Connor mentioned he's pushing some changes from #112.

ConorOkus commented:
Added the work from @Tiano01 [PR](https://github.com/BitcoinDesign/Guide/pull/112) on hiding sensitive information to the [protecting a wallet](https://deploy-preview-302--sad-borg-390916.netlify.app/guide/onboarding/protecting-a-wallet/#hiding-sensitive-information) section in onboarding, also made her a co-author in this [commit](https://github.com/BitcoinDesign/Guide/pull/302/commits/d9467aa3588655c203391e275518dc9947ccd627).

pavlenex commented:
Awesome @ConorOkus, I'll review the `Protecting a wallet page` next, which is the only thing left for me to review 🚀 gimme a day max.

pavlenex commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,223 @@
+
+Bitcoin wallets contain sensitive information such as recovery phrases, balances, transaction histories, and more. Therefore, it is crucial to add extra layers of security to keep private data out of bad actors' hands. Additional protection may also be necessary for users who share devices, such as parents with children.
+
+A common practice for wallets is to have a barrier (i.e. Touch ID, PINs, etc.) that must be unlocked to open a wallet application. Creating robust and challenging- to- crack passwords and PIN codes isn't something that everyone excels at. In order to decrease the chance of someone’s wallet getting hacked because of poor security decisions, you can implement the following into your design:
```
```suggestion
A common practice for wallets is to have a barrier (i.e., Touch ID, PINs, etc.) that must be unlocked to open a wallet application. Creating robust and challenging-to-crack passwords and PIN codes isn't something that everyone excels at. To decrease the chance of someone’s wallet getting hacked because of poor security decisions, you can implement the following into your design:
```

pavlenex commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,223 @@
+
+One convenient way for users to verify their identity to unlock your wallet app is with biometrics. This could take the form of either touch or face ID.
+
+Users of Android and iOS devices are super familiar with these motions. More importantly, using biometrics does not require the user to remember anything, furthermore decreasing their chances of getting locked out of their wallet.
```
```suggestion
Users of Android and iOS devices are already familiar with these motions. More importantly, using biometrics does not require the user to remember anything, decreasing their chances of getting locked out of their wallet.
```

pavlenex commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,223 @@
+
+### PIN
+
+A PIN is one of the most common and traditional methods for securing a mobile application. This setup requires that users create a 4-6 digit PIN that must be entered when opening their wallet application, or when viewing sensitive information such as their recovery phrase.
```
```suggestion
A PIN is one of the most common and traditional methods for securing a mobile application. This setup requires that users create a 4-6 digit PIN that must be entered when opening their wallet application or viewing sensitive information such as their recovery phrase.
```

pavlenex commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,223 @@
+
+</div>
+
+The PIN-based approaches can be supported by all devices, and therefore is more accessible. A PIN can also be used to encrypt data within a wallet, which makes any locally stored information on the device inaccessible. Additionally, some users prefer this type of security measure because they use the same or similiar PINs on multiple applications/devices, making it easy to remember.
```
```suggestion
All devices can support the PIN-based approach, which makes it more accessible. A PIN can also be used to encrypt data within a wallet, making any locally stored information on the device inaccessible. Additionally, some users prefer this type of security measure because they use the same or similar PINs on multiple applications/devices, making it easy to remember.
```

pavlenex commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,223 @@
+
+The PIN-based approaches can be supported by all devices, and therefore is more accessible. A PIN can also be used to encrypt data within a wallet, which makes any locally stored information on the device inaccessible. Additionally, some users prefer this type of security measure because they use the same or similiar PINs on multiple applications/devices, making it easy to remember.
+
+However, users with poor security practices might create easy-to-guess PINs (such as 1234), which would decrease the level of security. This design also relies on users to remember their PIN correctly.
```
```suggestion
However, users with poor security practices might create easy-to-guess PINs (such as 1234), which would provide a false sense of security. This design also relies on users to remember their PIN correctly.
```

pavlenex commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,223 @@
+
+{% include tip/open.html color="green" icon="check" label="Do: Educate on Good Security Practices" %}
+
+Within your product upon PIN creation, tell your users what a good, secure PIN is (i.e. not their birthday). Recommend against (or even make it impossible) to create a PIN such as 1234, 1111, etc.
```
```suggestion
Within your product upon PIN creation, tell your users what a good, secure PIN is (i.e., not their birthday). Recommend against (or even make it impossible) to create a PIN such as 1234, 1111, etc.
```

pavlenex commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,223 @@
+
+### Hiding Sensitive Information
+
+Imagine this scenario. You are in a public place, and you need to make a payment using your bitcoin wallet. You open your wallet on your phone, but you don’t feel comfortable having your address and balance information clearly visible to strangers who may be looking over your shoulder, persons lurking or video surveillance . Hence by giving users the ability to hide sensitive information in their ...
```
```suggestion
Imagine this scenario. You are in a public place, and you need to make a payment using your bitcoin wallet. You open your wallet on your phone, but you don’t feel comfortable having your address and balance information clearly visible to strangers who may be looking over your shoulder, persons lurking, or video surveillance. Hence by giving users the ability to hide sensitive information in their wallet if desired, they gain an added sense of physical privacy and security [^3]when using the app in public.
```

pavlenex commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,223 @@
+
+Imagine this scenario. You are in a public place, and you need to make a payment using your bitcoin wallet. You open your wallet on your phone, but you don’t feel comfortable having your address and balance information clearly visible to strangers who may be looking over your shoulder, persons lurking or video surveillance . Hence by giving users the ability to hide sensitive information in their ...
+
+### What information is considered sensitive ?
```
We already talk about what information is sensitive, private in the intro paragraph, perhaps merge in the information?

> Bitcoin wallets contain sensitive information such as recovery phrases, balances, transaction histories, and more. Therefore, it is crucial to add extra layers of security to keep private data out of bad actors' hands. Additional protection may also be necessary for users who share devices, such as parents with children.

Not a blocker, but could be improved to avoid repetition.

pavlenex commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,223 @@
+
+### What information is considered sensitive ?
+
+Sensitive information in wallet applications include the wallet balance, addresses, private keys and previous transactions information.
```
```suggestion
Sensitive information in wallet applications includes the wallet balance, addresses, private keys, and previous transactions information.
```

Bosch-0 commented on guide/onboarding/restoring-a-wallet.md:
```diff
@@ -0,0 +1,51 @@
+
+As outlined earlier in the chapter, we consider an automatic cloud backup a good option, especially for beginners. One of the major UX benefits of this is a seamless recovery process. All users needs to do to gain access to their funds again is log into their Google or Apple account.
+
+If they aren't logged in, they only need to enter their password before the wallet can automatically access the recovery phrase stored on the user's iCloud or Google Drive account. Once the recovery phrase is located, the wallet can begin to regenerate the keys required for the user to access their funds.
```
I think creating a standard way to do cloud backups was the aim of photon. Standard way to do this does not currently exist. https://github.com/photon-sdk

pavlenex commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,223 @@
+- Addresses - can be used to track your transaction history
+- Private keys - can be used to access and transfer your bitcoins
+
+It's more common for wallets to protect private keys but not much is done for other sensitive information like the balance, addresses and previous transactions. A few wallets like Bitcoin Core, Wasabi, Wallet of Satoshi, and others have made it work though. Below describes a pattern, and considerations for hiding and revealing sensitive information.
```
```suggestion
It's more common for wallets to protect private keys, but not much is done for other sensitive information like the balance, addresses, and previous transactions. A few wallets like Bitcoin Core, Wasabi, Wallet of Satoshi, and others have made it work, though. Below is a pattern and considerations for hiding and revealing sensitive information.
```

pavlenex commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,223 @@
+
+### Quickly hide from overview screen
+
+The hide icon / button, which is usually displayed within close reach of the balance itself, is used to quickly and easily hide and reveal wallet information by tapping or clicking on it.[^2]
```
```suggestion
The hide icon/button, which is usually displayed within close reach of the balance itself, is used to quickly and easily hide and reveal wallet information by tapping or clicking on it.[^2]
```

pavlenex commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,223 @@
+- Easy for anyone else to reveal your information if they have access to your device.
+
+
+## Entering a PIN to reveal information
```
Does this slightly feel like a duplication of the "Pin" section?

ConorOkus replied:
Yeah could argue that. I think it's okay though as this provides a more specific case for revealing a users balance. Initially is more for just locking the whole app.

pavlenex replied:
Agreed, not really a blocker for the current phase for sure.

pavlenex commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,223 @@
+
+###  Hide if inactive
+
+Another solution would be to invoke the wallet’s hidden state as a default when the app is opened. What this means is that in the event that someone sees your wallet when you open it, they wouldn't be able to make out the information. The pre-hidden state can be unveiled  after a tap, PIN entry, or perhaps a short 5 second timer.
```
```suggestion
Another solution would be to invoke the wallet’s hidden state as a default when the app is opened. This means that if someone sees your wallet when you open it, they wouldn't be able to make out the information. The pre-hidden state can be unveiled after a tap, PIN entry, or perhaps a short 5-second timer.
```

pavlenex commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,223 @@
+
+### Making the toggle less obvious
+
+Having the show/hide button right on the main screen make things quite obvious for someone who has access to your device to press unhide. A solution would be to move the hide toggle away from the home screen and into the app settings[^1]. This way, if someone has access to your device and opens the app, they may not immediately know how to reveal the balance, transaction and/or addresses as it is ...
```
```suggestion
Having the show/hide button right on the main screen makes things quite obvious for someone who has access to your device to press unhide. A solution would be to move the hide toggle away from the home screen and into the app settings[^1]. This way, if someone has access to your device and opens the app, they may not immediately know how to reveal the balance, transaction and/or addresses as it is not made obvious as the previous solutions.
```

pavlenex reviewed (commented):
Reviewed the final page that I had left, `Protecting a wallet.`
Great work, @ConorOkus.
I think Tiana's section is well-suited here. There maybe are certain repetitions in the sections, such as PIN and Intro sections. However, they convey information well. If there's no time to re-write these, I don't see them as a big blocker, but it would be nice to try to unify them slightly to avoid repetitiveness.

The visuals are spot-on. I've found a few things that can be addressed below.

GBKS commented on guide/onboarding/backing-up-a-recovery-phrase.md:
```diff
@@ -0,0 +1,44 @@
+
+# Backing up a recovery phrase
+
+When a user generates a new wallet, it creates something called a private key (also known as a recovery phrase). A [recovery phrase]({{ 'guide/glossary/#recovery-phrase' | relative_url }}), is a group of 12 or 24 words that let users recover funds if wallet access is lost. Likewise, users who lose both their recovery phrase and wallet access permanently lose their bitcoin. Arguably, this creates  ...
```
> it creates something called a private key

I don't think recovery phrase is called private key, is it?

ConorOkus replied:
🚧

GBKS commented on guide/onboarding/backing-up-a-recovery-phrase.md:
```diff
@@ -0,0 +1,44 @@
+   image = "/assets/images/guide/onboarding/backing-up-a-recovery-phrase/verify-recovery-phrase.png"
+   retina = "/assets/images/guide/onboarding/backing-up-a-recovery-phrase/verify-recovery-phrase@2x.png"
+   width = 250
+   height = 250
```
Incorrect height. The image is not square.

GBKS commented on guide/onboarding/backing-up-a-recovery-phrase.md:
```diff
@@ -0,0 +1,44 @@
+   width = 250
+   height = 250
+   alt-text = "Recovery phrase interface"
+   caption = "Typical recovery phrase UI from the [UI Kit](Bitcoin Wallet UI Kit)"
```
Broken link.

GBKS commented on guide/onboarding/backing-up-a-recovery-phrase.md:
```diff
@@ -0,0 +1,44 @@
+
+However, there may be times when it is necessary to use a manual backup alternative, as it can be low tech and easily accessible.
+
+The next sections will look at some UI considerations for [automatic cloud backups]({{ 'guide/onboarding/backing-up-recovery-phrase/automatic-cloud-backup' | relative_url }}) and [manual backups]({{ 'guide/onboarding/backing-up-recovery-phrase/manual-backup' | relative_url }}).
```
Both links are broken.

GBKS commented on guide/onboarding/automatic-cloud-backup.md:
```diff
@@ -0,0 +1,103 @@
+
+### How it works
+
+The user’s key is encrypted and backed up on their respective device/cloud storage provider. If their device is lost, stolen, or broken, the user can log into their Apple or Google account to regain access to their funds. To learn more about the technical details, hop over to the [private key management]({{ '/guide/private-key-management/introduction' | relative_url }}) section on this topic.
```
```suggestion
The user’s key is encrypted and backed up on their respective device/cloud storage provider. If their device is lost, stolen, or broken, the user can log into their cloud storage account (such as Apple iCloud or Google Drive) to regain access to their funds. To learn more about the technical details, hop over to the [private key management]({{ '/guide/private-key-management/introduction' | relative_url }}) section on this topic.
```

GBKS commented on guide/onboarding/automatic-cloud-backup.md:
```diff
@@ -0,0 +1,103 @@
+
+{% include tip/open.html color="green" icon="check" label="Do: Inform your Users" %}
+
+Before automatically backing up a recovery phrase via cloud backup, make sure the user knows what they are about to do. Once the backup is complete, provide a summary and some reassurances before moving on. This is a good place for loading interactions and animation.
```
> This is a good place for loading interactions and animation.

What do you mean by that? Why is it a good place?

ConorOkus replied:
This is because of the time delay between when users click to upload and when it completes. 

Will update 🚧

GBKS commented on guide/onboarding/manual-backup.md:
```diff
@@ -0,0 +1,264 @@
+
+### When to reveal a recovery phrase
+
+The onboarding carousels that we previously mentioned are a great place to prep and explain to users that they will be given a recovery phrase. While some wallets skip this step, in might be appropriate in some scenarios to display it early on. Doing so in the initial onboarding process allows users to pause, internalize what a recovery phrase is, and ensures that they do not neglect this step la ...
```
> the onboarding carousels that we previously mentioned

No idea where this was previously mentioned. I'd either make it a link to make me to that mention if there's useful information, or shorten it to `Onboarding carousels are a great...`.

GBKS commented on guide/onboarding/manual-backup.md:
```diff
@@ -0,0 +1,264 @@
+
+<div class="center" markdown="1">
+
+{% include image.html
```
Yep, correct image is [here](https://www.figma.com/file/VB3GQdAnhl8yta44DY3PSV/Bitcoin-Wallet-UI-Kit?node-id=1230%3A131).

GBKS commented on guide/onboarding/restoring-a-wallet.md:
```diff
@@ -0,0 +1,51 @@
+
+### Restore from an automatic cloud backup
+
+As outlined earlier in the chapter, we consider an automatic cloud backup a good option, especially for beginners. One of the major UX benefits of this is a seamless recovery process. All users needs to do to gain access to their funds again is log into their Google or Apple account.
```
```suggestion
As outlined earlier in the chapter, we consider an automatic cloud backup a good option, especially for beginners. One of the major UX benefits of this is a seamless recovery process. All users needs to do to gain access to their funds again is to log into their cloud storage account.
```

GBKS commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,223 @@
+   height = 800
+%}
+
+#### Pros
```
I'd just use plain text instead of Pros and Cons on this page. Each technique just has one pro or con, and they are each just one sentence. No need for lots of bullet lists that contain just one item.

GBKS commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,223 @@
+   mobile = "/assets/images/guide/onboarding/protecting-a-wallet/hide-by-eye-icon.gif"
+   mobileRetina = "/assets/images/guide/onboarding/protecting-a-wallet/hide-by-eye-icon.gif"
+   alt-text = "hide-by-eye icon"
+   width = 1600
```
Image size are wrong on these GIFs.

GBKS replied:
No need for the retina, mobile and mobileRetina attributes if they all specify the same image.

GBKS replied:
Also needs proper alt text, not just the file name.

ConorOkus replied:
🚧

GBKS commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,223 @@
+- Addresses - can be used to track your transaction history
+- Private keys - can be used to access and transfer your bitcoins
+
+It's more common for wallets to protect private keys, but not much is done for other sensitive information like the balance, addresses, and previous transactions. A few wallets like Bitcoin Core, Wasabi, Wallet of Satoshi, and others have made it work, though. Below is a pattern and considerations for hiding and revealing sensitive information.
```
Link out to those referenced wallets?

GBKS commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,223 @@
+- Easy for anyone else to reveal your information if they have access to your device.
+
+
+## Entering a PIN to reveal information
```
The header hierarchy is off here. This header should probably not be H2 but something lower.

GBKS commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,223 @@
+
+## Entering a PIN to reveal information
+
+A problem that seemed to arise throughout each implementation was that it was as easy to reverse the hidden state as it was to enable it. This may be for convenience sake, however, for protection against unauthorized access, perhaps you should only be able to unhide your information if a PIN or password has been entered. This could therefore reaffirm the identity of the wallet owner for extra sec ...
```
> A problem that seemed to arise throughout each implementation...

What is meant by that? Who was implementing what? Maybe some outdated text that should be removed?

The second sentence is also weird with the reference to `you`.

GBKS commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,223 @@
+
+#### Cons
+
+- It may not be convenient for the users to repeatedly put in their PIN when ever they want to reveal your information especially if they do so often.
```
```suggestion
- It may not be convenient for users to repeatedly enter their PIN to reveal sensitive information, especially if they do this often.
```

GBKS commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,223 @@
+
+#### Pros
+
+- The risk of an unauthorized person revealing your information is minimal due to the PIN required
```
```suggestion
- The risk of an unauthorized person revealing a users information is minimal due to the PIN requirement
```

GBKS commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,223 @@
+
+###  Hide if inactive
+
+Another solution would be to invoke the wallet’s hidden state as a default when the app is opened. This means that if someone sees your wallet when you open it, they wouldn't be able to make out the information. The pre-hidden state can be unveiled after a tap, PIN entry, or perhaps a short 5-second timer.
```
```suggestion
Another solution is to invoke the wallet’s hidden state as a default when the app is opened, to protect against prying eyes during initial display. The pre-hidden state can be unveiled after a tap, PIN entry, or perhaps a short 5-second timer.
```

GBKS commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,223 @@
+
+#### Pros
+
+- Users have some time to assess their environment before their info is displayed.
```
```suggestion
- Users have time to assess their environment before their info is displayed.
```

GBKS commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,223 @@
+
+#### Cons
+
+- The user may feel a bit frustrated having to wait for their information to be revealed especially in an urgent situation
```
```suggestion
- Users may get frustrated having to wait for their information to be revealed, especially in urgent situations
```

GBKS commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,223 @@
+
+### Making the toggle less obvious
+
+Having the show/hide button right on the main screen makes things quite obvious for someone who has access to your device to press unhide. A solution would be to move the hide toggle away from the home screen and into the app settings[^1]. This way, if someone has access to your device and opens the app, they may not immediately know how to reveal the balance, transaction and/or addresses as it i ...
```
```suggestion
Having the show/hide button right on the main screen makes it quite obvious for someone who has access to a users device to reveal the information. A solution would be to move the button away from the home screen and into the app settings[^1]. This way, if someone has access to a users device and opens the app, they may not immediately know how to reveal the balance, transaction and/or addresses, as it is not made obvious as in the previous solutions.
```

GBKS commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,223 @@
+
+#### Pros
+
+- The risk of an unauthorized person revealing your information is minimal due to the fact that the toggle isn't immediately visible on the home screen
```
```suggestion
- The risk of an unauthorized person revealing a users information is minimal due to the fact that the toggle isn't immediately visible on the home screen
```

GBKS commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,223 @@
+
+#### Cons
+
+- This solution may not be convenient for a user to quickly hide their information if the need arises.
```
```suggestion
- This solution may not be convenient for users to quickly hide their information if the need arises
```

GBKS commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,223 @@
+
+Privacy in bitcoin payments goes far beyond hiding balances and other sensitive information, the privacy by design framework[^4] states that privacy should be incorporated and built into products by default. This way, whether or not the user is concerned with their data privacy, they would always be protected through good UX and UI.
+
+By including the hiding information pattern in design of wallets, we give users a greater sense of control and comfortability in any environment when using it. They have the freedom to decide whether or not they want their information visible.
```
```suggestion
By including UX patterns for hiding information pattern wallets, we give users a greater sense of control and comfort in any situation. They have the freedom to decide whether or not they want their information visible.
```

GBKS commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,223 @@
+
+Now let's look at the different ways a user might go about [funding a wallet]({{ 'guide/onboarding/funding-a-wallet' | relative_url }}).
+
+[^1]: https://medium.com/@olanrewajusodiq64/the-ui-ux-of-hide-balance-designing-to-improve-asset-security-e4b20668f315
```
There's no need for footnotes on this page. Should be just links.

GBKS commented on guide/onboarding/funding-a-wallet.md:
```diff
@@ -0,0 +1,91 @@
+   width = 250
+   height = 250
+   alt-text = "Showing recovery phrase one word at a time"
+   caption = "Buy bitcoin screen from the from [Wallet UI Kit](https://www.figma.com/file/VB3GQdAnhl8yta44DY3PSV/Bitcoin-Wallet-UI-Kit?node-id=1228%3A28550)"
```
```suggestion
   caption = "Buy bitcoin screen from the [Wallet UI Kit](https://www.figma.com/file/VB3GQdAnhl8yta44DY3PSV/Bitcoin-Wallet-UI-Kit?node-id=1228%3A28550)"
```

GBKS commented on guide/onboarding/funding-a-wallet.md:
```diff
@@ -0,0 +1,91 @@
+
+{% include image.html
+   image = "/assets/images/guide/onboarding/funding-a-wallet/buy-bitcoin.png"
+   retina = "/assets/images/guide/onboarding/funding-a-wallet/buy-bitcoin.png"
```
Add a real retina image.

GBKS commented on guide/onboarding/funding-a-wallet.md:
```diff
@@ -0,0 +1,91 @@
+   image = "/assets/images/guide/onboarding/funding-a-wallet/buy-bitcoin.png"
+   retina = "/assets/images/guide/onboarding/funding-a-wallet/buy-bitcoin.png"
+   width = 250
+   height = 250
```
Height is wrong, image is not square.

GBKS commented on guide/onboarding/funding-a-wallet.md:
```diff
@@ -0,0 +1,91 @@
+   image = "/assets/images/guide/onboarding/funding-a-wallet/bitcoin-gift-cards.png"
+   retina = "/assets/images/guide/onboarding/funding-a-wallet/bitcoin-gift-cards@2x.png"
+   alt-text = "Bitcoin gift cards"
+   caption = "Bitcoin gift cards"
```
```suggestion
   caption = "Bitcoin gift card concept"
```

GBKS commented:
Just read through everything and left comments. Overall great, just a lot of small things to fix, which is to be expected with so many new pages. 

A few notes:
- Make sure to run `rake test`, there are various broken links
- Please check that images have proper retina images, correct widths and heights and good alt-texts and captions
- Do you have plans for getting the remaining header images done?
- The "Hiding sensitive information" content is good, but it reads a bit shaky (like flipping back and forth between "users wallet" and "your wallet"). I added some comments, but might need a few more iterations
- Should "Protecting a wallet" and "Funding a wallet" be part of onboarding?

There is a [bunch of back and forth](https://github.com/BitcoinDesign/Guide/pull/319) around the naming of the "Automatic cloud backup" and "Manual backup" page titles in the private key management section. As this section also has pages with those names, we need to ensure we keep things consistent, whatever the final names will be.

There's a lot of nuance in these pages that we need to make sure we get right. For example, automatic cloud backup does not have to involve a recovery phrase (like in Muuns case) - should we point that out? That page also goes back and forth between the term "keys" and "recovery phrase" - is that OK?

sbddesign commented on guide/onboarding/manual-backup.md:
```diff
@@ -0,0 +1,264 @@
+
+{% include /tip/open.html label="Tip: Be Clear about Numbering" icon="info" color="blue" %}
+
+Regardless of the option you decide to run with, it’s important that you explicitly instruct users to number each word (i.e. 1. sand 2. purple 3. flower). Manually backing up recovery phrases typically asks users to recount, for example, the fifth word. It’s a bit of a pain for users to count which word corresponds to a particular number if they are not numbered initially.
```
```
Regardless of the option you decide to run with, it’s important that you explicitly instruct users to number each word (i.e. 1. sand 2. purple 3. flower). Typically, a manual recovery phrase backup scheme will ask users to recount, for example, the fifth word. It’s a bit of a pain for users to count which word corresponds to a particular number if they are not numbered initially.
```

sbddesign replied:
The sentence might have been grammatically correct, but somethings sounds awkward to me about it.

sbddesign commented on guide/onboarding/manual-backup.md:
```diff
@@ -0,0 +1,264 @@
+
+A clever way to give a user their recovery phrase is by creating an interaction where they have to “uncover” it. This can entail hovering over a blurred word to reveal it, or holding down a button to show it.
+
+By creating this design interaction, it drills in how, viewing, storing, and managing recovery phrases should be a private and secret matter.
```
```suggestion
By creating this design interaction, it drills in how viewing, storing, and managing recovery phrases should be a private and secret matter.
```

sbddesign commented on guide/onboarding/restoring-a-wallet.md:
```diff
@@ -0,0 +1,50 @@
+
+<div class="center" markdown="1">
+
+There are a number of reasons why a user might need to recover their wallet. Typically, this might be because they lost their device, had it stolen, are migrating from a different wallet or deleted and redownloaded a particular wallet application. Regardless of how it occured, it’s important that your wallet application has clear, easy instructions that provide an easy way for users to recover th ...
```
```suggestion
There are a number of reasons why a user might need to recover their wallet. Typically, this might be because they lost their device, had it stolen, are migrating from a different wallet or deleted and redownloaded a particular wallet application. Regardless of how it occurred, it’s important that your wallet application has clear, easy instructions that provide an easy way for users to recover their funds. There are a number of ways that a wallet recovery can be done. However, as covered in the [wallet interoperability]({{ 'guide/foundations/wallet-interoperability' | relative_url }}) section, not all wallets from one application can be recovered in another.
```

sbddesign commented on guide/onboarding/protecting-a-wallet.md:
```diff
@@ -0,0 +1,175 @@
+
+## Why are these patterns important to designers?
+
+Privacy in bitcoin payments goes far beyond hiding balances and other sensitive information, the [privacy by design framework](https://www.ipc.on.ca/wp-content/uploads/Resources/7foundationalprinciples.pdf) states that privacy should be incorporated and built into products by default. This way, whether or not the user is concerned with their data privacy, they would always be protected through go ...
```
```suggestion
Privacy in bitcoin payments goes far beyond hiding balances and other sensitive information. The [privacy by design framework](https://www.ipc.on.ca/wp-content/uploads/Resources/7foundationalprinciples.pdf) states that privacy should be incorporated and built into products by default. This way, whether or not the user is concerned with their data privacy, they would always be protected through good UX and UI.
```

sbddesign commented on guide/onboarding/manual-backup.md:
```diff
@@ -0,0 +1,264 @@
+
+The onboarding carousels that we previously mentioned are a great place to prep and explain to users that they will be given a recovery phrase. While some wallets skip this step, in might be appropriate in some scenarios to display it early on. Doing so in the initial onboarding process allows users to pause, internalize what a recovery phrase is, and ensures that they do not neglect this step la ...
+
+{% include /tip/open.html label="Tip: Let users transact even when not backed up" icon="info" color="blue" %}
+
+If you really want to skip showing a user their recovery phrase during the initial onboarding process, create an interaction (i.e. a vibration, modal pop-up, or blocking users from clicking anything) that prompts users to “back up” their recovery phrase at a later time. Perhaps before making any transactions or after several transactions.
```
I think I get the general idea, but the phrasing is a little ambiguous. Is it suggesting to *let* users transact when not backed up? Or is it suggesting to *block* users from transacting when not backed up? I feel like the header and paragraph contradict each other.

ConorOkus replied:
Yeah it's implying you should let users transact regardless, I'll remove the stuff about blocking users. Thanks 👍

sbddesign commented on guide/onboarding/automatic-cloud-backup.md:
```diff
@@ -0,0 +1,103 @@
+   height = 541
+%}
+
+{% include picture.html
+     image = "/assets/images/guide/onboarding/backing-up-a-recovery-phrase/automatic-cloud-backup/cloud-backup-password.png"
+   retina = "/assets/images/guide/onboarding/backing-up-a-recovery-phrase/automatic-cloud-backup/cloud-backup-password@2x.png"
+   class = "shadow"
+   alt-text = "The user can set a password which will be used to encrypt their recovery phrase and act as a login"
+   width = 250
+   height = 541
+%}
```
I am struggling with this. On the one hand, it makes absolute sense that you want your recovery seed to be password protected. On the other hand, I don't feel like this solves any problems for a new user. How is this any different from 24 seed words? If they are capable of losing their seed words, it seems to me that they are equally as likely to lose the passphrase which encrypts their seed words.

I think there are ways around this (Casa has some scheme for it I believe).

I don't intend to over-complicate this, it just seems to me like a recovery phrase and a passphrase are different ways of describing the same thing: something that a new user will ultimately lose.

Is there a detail I am missing?

ConorOkus replied:
This is a good point.

In practice rather than using a passphrase or password, a simple 4 to 6 digit PIN would be sufficient to gain access to the wallet, it's easier to remember and the user is less likely to worry about having to back this up. 

In practice, this is what we have been experimenting with on the [Photon SDK ](https://github.com/photon-sdk/photon-lib) and it works well. 

I'll replace the images of the password screen with a PIN-based one as this is more in line with what we would want to advocate for. Keep in mind this scheme is for what a user would consider low-value amounts.

sbddesign reviewed (commented):
Loving it! I left some suggestions on specific lines.

pavlenex reviewed (approved):
Took one last look at this, and it looks like it's ready to go. tACK.

@ConorOkus if there's something more you'd like to add let us know, otherwise let's give a final glance at this and get it merged in since it's a big and critical PR.

Thanks, Conor for going through all the suggestions and implementing them. 🚀

ConorOkus commented:
> Just read through everything and left comments. Overall great, just a lot of small things to fix, which is to be expected with so many new pages.
> 
> A few notes:
> 
> * Make sure to run `rake test`, there are various broken links
> * Please check that images have proper retina images, correct widths and heights and good alt-texts and captions
> * Do you have plans for getting the remaining header images done?
> * The "Hiding sensitive information" content is good, but it reads a bit shaky (like flipping back and forth between "users wallet" and "your wallet"). I added some comments, but might need a few more iterations
> * Should "Protecting a wallet" and "Funding a wallet" be part of onboarding?
> 
> There is a [bunch of back and forth](https://github.com/BitcoinDesign/Guide/pull/319) around the naming of the "Automatic cloud backup" and "Manual backup" page titles in the private key management section. As this section also has pages with those names, we need to ensure we keep things consistent, whatever the final names will be.
> 
> There's a lot of nuance in these pages that we need to make sure we get right. For example, automatic cloud backup does not have to involve a recovery phrase (like in Muuns case) - should we point that out? That page also goes back and forth between the term "keys" and "recovery phrase" - is that OK?

Looking to do some work with Jessica Wolff for the banner images that are missing on some of these pages. Unsure if they will be done by launch but I don't think we view this as a blocker to releasing.

I've addressed the content in the "Hiding sensitive information" to speak to the reader more.

I think `Protecting a wallet` and `Funding a wallet` are suitable to be in the onboarding sections because giving guidance on protecting a wallet allows the user the best chance to make a transaction safely and successfully early on. Also, understanding funding options will give product teams a more holistic idea of how someone might be interacting with Bitcoin outside of their specific app.  

Will leave some comments in the other PR related to auto vs manual backups.

ConorOkus commented on guide/onboarding/manual-backup.md:
```diff
@@ -0,0 +1,264 @@
+
+{% include /tip/close.html %}
+
+### Ways to show it
```
Good shout 🚧

Bosch-0 commented on guide/onboarding/automatic-cloud-backup.md:
```diff
@@ -95,7 +95,7 @@ By not making users manually backup, you decrease the risk of self-inflicted los
 While this is an easier way for users to set up a wallet, it brings in the potential threat of malicious third-party access. If a user’s cloud account is hacked or they forget their cloud login information, their funds could be compromised.
 
-Additionally, by bringing cloud storage providers into the equation, we assume that users trust their providers. This also raises the question of [decentralization]({{ 'guide/foundations/principles/#decentralization' | relative_url }}): a guiding principle of the bitcoin space and could be considered at conflict with the popular saying, “not your keys, not your coins”.
+By bringing cloud storage providers into the equation, we assume that users trust their providers. This also raises the question of [decentralization]({{ 'guide/foundations/principles/#decentralization' | relative_url }}): a guiding principle of the Bitcoin space and could be considered in conflict with the popular saying, “not your keys, not your coins”.
```
I know photon is kind of on hold but could it recommended to designers to integrate some kind of encryption on the data stored in the cloud? Ideally by a users entered PIN as that is much harder to forgot than a passphrase.

ConorOkus replied:
I updated the gallery image to use PIN-based auth 👍

pavlenex commented on guide/onboarding/funding-a-wallet.md:
```diff
@@ -0,0 +1,91 @@
+>
+> <cite>As reported by <a href="https://www.nbcsports.com/northwest/seahawks/former-seattle-seahawk-russell-okung-puts-half-salary-bitcoin-considered-highest">NBC Sports</a></cite>
+
+#### Pros
+ - Privacy
+ - No exchange fees
+ - Simplifies onboarding
```
For visual consistency with other chapters.
```suggestion
{% include fact/pros.html %}
 - Privacy
 - No exchange fees
 - Simplifies onboarding
 {% include fact/close.html %}
```

pavlenex commented on guide/onboarding/funding-a-wallet.md:
```diff
@@ -0,0 +1,91 @@
+ - No exchange fees
+ - Simplifies onboarding
+
+#### Cons
+ - Managing taxes
+ - Difficult to find businesses/individuals willing to pay in bitcoin
+ - Limited options for recurring payments
```
Visal consistency.

```suggestion
{% include fact/cons.html %}
 - Managing taxes
 - Difficult to find businesses/individuals willing to pay in bitcoin
 - Limited options for recurring payments
 {% include fact/close.html %}
```

pavlenex commented on guide/onboarding/funding-a-wallet.md:
```diff
@@ -0,0 +1,91 @@
+
+{% include /tip/close.html %}
+
+#### Pros
+ - An easy and popular way to acquire bitcoin
```
```suggestion
{% include fact/pros.html %}
  - An easy and popular way to acquire bitcoin
{% include fact/close.html %}

```

pavlenex commented on guide/onboarding/funding-a-wallet.md:
```diff
@@ -0,0 +1,91 @@
+#### Pros
+ - An easy and popular way to acquire bitcoin
+
+#### Cons
+ - Managing taxes
+ - Difficult to find businesses/individuals willing to pay in bitcoin
+ - Hard to set up recurring payments
+ - Not private
```
```suggestion
{% include fact/cons.html %}
 - Managing taxes
 - Difficult to find businesses/individuals willing to pay in bitcoin
 - Hard to set up recurring payments
 - Not private
 {% include fact/close.html %}

```

pavlenex commented on guide/onboarding/funding-a-wallet.md:
```diff
@@ -0,0 +1,91 @@
+   height = 500
+%}
+
+#### Pros
+ - Gets user comfortable with scanning QR codes
+ - Simplifies onboarding
+ - No compromise on privacy
```
```suggestion
{% include fact/pros.html %}
 - Gets user comfortable with scanning QR codes
 - Simplifies onboarding
 - No compromise on privacy
 {% include fact/close.html %}
```

pavlenex commented on guide/onboarding/funding-a-wallet.md:
```diff
@@ -0,0 +1,91 @@
+ - Simplifies onboarding
+ - No compromise on privacy
+
+#### Cons
+ - Low retail availability
```
```suggestion
{% include fact/close.html %}
 - Low retail availability
 {% include fact/close.html %}
```

pavlenex reviewed (dismissed):
LGTM, just saw we don't use pros/cons visually consistent, so added suggestions for those.

pavlenex commented on guide/onboarding/funding-a-wallet.md:
```diff
@@ -0,0 +1,99 @@
+ - No compromise on privacy
+ {% include fact/close.html %}
+
+{% include fact/close.html %}
+ - Low retail availability
+ {% include fact/close.html %}
```
```suggestion
{% include fact/close.html %}
 - Low retail availability
{% include fact/close.html %}
```

pavlenex commented on guide/onboarding/funding-a-wallet.md:
```diff
@@ -0,0 +1,99 @@
+   height = 500
+%}
+
+{% include fact/pros.html %}
+ - Gets user comfortable with scanning QR codes
+ - Simplifies onboarding
+ - No compromise on privacy
+ {% include fact/close.html %}
```
```suggestion
{% include fact/pros.html %}
 - Gets user comfortable with scanning QR codes
 - Simplifies onboarding
 - No compromise on privacy
{% include fact/close.html %}
```

pavlenex reviewed (approved):
Merging this as previously agreed as the PR is quite big and critical for us to proceed on other PR's. I've pushed a fix for formatting in https://github.com/BitcoinDesign/Guide/pull/302/commits/85bf62e8a6af6372f1e1bdf705e96700981c6538, which was my bad when adding a suggestion.

Great work. Let's address any minor stuff (if any) in smaller PR's in the future. 🚀 🚀 🚀
